master/pkg/model/experiment_config.go

```diff
@@ -184,6 +186,33 @@ func (l *Labels) UnmarshalJSON(data []byte) error {
 	return err
 }
 
+// DevicesConfig is the configuration for bind mounts.
+type DevicesConfig []DeviceConfig
+
+// UnmarshalJSON implements the json.Unmarshaler interface so that DeviceConfigs are additive.
+func (d *DevicesConfig) UnmarshalJSON(data []byte) error {
+	unmarshaled := make([]DeviceConfig, 0)
+	if err := json.Unmarshal(data, &unmarshaled); err != nil {
+		return errors.Wrap(err, "failed to parse devices")
+	}
+	*d = append(*d, unmarshaled...)
+	return nil
+}
+
+// DeviceConfig configures trial runner filesystem bind mounts.
+type DeviceConfig struct {
+	HostPath      string `json:"host_path"`
+	ContainerPath string `json:"container_path"`
+	Mode          string `json:"mode"`
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (d *DeviceConfig) UnmarshalJSON(data []byte) error {
+	d.Mode = "mrw"
```

Comment:
obviously with all of this, it would be nice if there was a separate defaulting step but I think we can cross that bridge when we get there. this works just fine as far as i'm concerned for now.

Reply:
I don't think there's another way to do the defaulting for entire lists of objects with our current system.  There will be with expconf though; but this is simply a limitation of the json-unmarshal-into-defaults strategy.

Reply:
yeah I totally agree. we should do this like expconf one day, but today isn't the day.